src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -120,6 +120,22 @@ object Simplify {
 
       case GetTupleElement(MakeTuple(xs), idx) => xs(idx)
 
+      // lift up NAs
+      case x@Apply(_, args) if args.exists(_.isInstanceOf[NA]) => NA(x.typ)
```

Comment:
So I think these can all just be `case x: IR if isStrict(x) and x.children.exists(_.isInstanceOf[NA]) => NA(x.typ)`.

Reply:
And that rule already exists:

```
      // propagate NA
      case x: IR if isStrict(x) && Children(x).exists(_.isInstanceOf[NA]) =>
        NA(x.typ)
```

We might need to add some IR to `isStrict`.  Alternatively, if some IR are strict only in some arguments, then we should have `isStrict(x, i)` which is true if `x` is strict in child `i`.

Reply:
aha.